### PR TITLE
Added DistributedOptimizerExt which supports virtually increasing batch size for TensorFlow

### PR DESCRIPTION
**Pull request by fps7806**
DistributedOptimizerExt allows simulation of a larger batch size by accumulating gradients over many iterations.
It extends horovod.tensorflow.DistributedOptimizer which allows multiplying the batch_size in a distributed setting.

Notes:
The class will wait until apply_gradients is called to accumulate gradients (rather than doing so in compute_gradients).

### CODE REVIEW

**Comment by alsrgv** [comment]
Notes from before:
1) We can either make the user call compute_gradients/aggregate_gradients N times before calling apply_gradients.
2) Or apply_gradients only actually apply_gradients every N times it is called

Option 1 makes the user do extra work and it's more aligned with PyTorch. Option 2 is not super intuitive.
